bundles/core/org.eclipse.smarthome.core.voice/src/main/java/org/eclipse/smarthome/core/voice/internal/VoiceManagerImpl.java

```diff
@@ -139,20 +141,24 @@ public void say(String text, String voiceId, String sinkId) {
             } else if (voiceId.contains(":")) {
                 // it is a fully qualified unique id
                 String[] segments = voiceId.split(":");
-                tts = ttsServices.get(segments[0]);
-                voice = getVoice(tts.getAvailableVoices(), segments[1]);
+                tts = getTTS(segments[0]);
```

Comment:
I changed it for readability and to prevent NPE (in case of `segment[0]` contains a not suitable value). Have a look at [https://github.com/eclipse/smarthome/pull/3764/files#diff-2bdbf6176fb6bd9b1af9a6cb230b6293R472](https://github.com/eclipse/smarthome/pull/3764/files#diff-2bdbf6176fb6bd9b1af9a6cb230b6293R472). The `getTTS(String)` method returns exaclty the same like before (`ttsServices.get(id)`).